AutoCollection/HttpRequestParser.ts

```diff
@@ -204,23 +247,27 @@ class HttpRequestParser extends RequestParser {
         if (request.headers) {
             this.correlationContextHeader = request.headers[RequestResponseHeaders.correlationContextHeader];
 
-            if (request.headers[RequestResponseHeaders.requestIdHeader]) {
-                this.parentId = request.headers[RequestResponseHeaders.requestIdHeader];
-                this.requestId = CorrelationIdManager.generateRequestId(this.parentId);
-                this.correlationContextHeader = request.headers[RequestResponseHeaders.correlationContextHeader];
+            if (CorrelationIdManager.w3cEnabled) {
+                this.parseW3CTraceHeaders(request, requestId);
             } else {
-                // Legacy fallback
-                const rootId = request.headers[RequestResponseHeaders.rootIdHeader];
-                this.parentId = request.headers[RequestResponseHeaders.parentIdHeader];
-                this.requestId = CorrelationIdManager.generateRequestId(rootId || this.parentId);
-                this.correlationContextHeader = null;
-            }
-            if (requestId) {
-                // For the scenarios that don't guarantee an AI-created context,
-                // override the requestId with the provided one.
-                this.requestId = requestId;
+                if (request.headers[RequestResponseHeaders.requestIdHeader]) {
+                    this.parentId = request.headers[RequestResponseHeaders.requestIdHeader];
+                    this.requestId = CorrelationIdManager.generateRequestId(this.parentId);
+                    this.correlationContextHeader = request.headers[RequestResponseHeaders.correlationContextHeader];
+                } else {
+                    // Legacy fallback
+                    const rootId = request.headers[RequestResponseHeaders.rootIdHeader];
```

Comment:
Are these doubly-old headers still read in w3cEnabled mode?

Reply:
They should still be read, but seem to not be read as it is now. I will re-arrange some stuff so that w3c mode is run in addition existing logic, instead of completely replacing it

Reply:
Resolving this as I refactored it bit so that code looks cleaner. Each parsing scenario is now split into its own segment. The bit where the generated Traceparent saves back-compat info is shared between all w3c parsing segments.